### PR TITLE
Move image to client (fixes #31)

### LINKED ISSUE #31: Move product icon to client

### PR DESCRIPTION
**Pull request by muffinresearch**
See also:
* mozilla/payments-example#20
* mozilla/payments-ui#140

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -27,8 +27,19 @@ function PaymentsClient(config) {
...
-  if (typeof this.product !== 'string') {
-    throw new Error('A \'product\' string must be provided');
+  if (typeof config.product !== 'object' || !config.product.id) {
+    throw new Error('A product id must be provided in a product object');
+  }
+
+  var productImageUrl = config.product.image;
+  if (!productImageUrl) {
+    console.warn(
+      'A product image URL was not supplied as a property of config.product.');
```
Comment (andymckay) [inline src/index.js]: Can we fall back to the Firefox logo if no image is supplied? It will look wierd without an image.

File: src/index.js [hunk 2]
```diff
@@ -27,8 +27,19 @@ function PaymentsClient(config) {
...
+  } else {
+    var imageProtocol = utils.getProtocol(productImageUrl);
+    if (imageProtocol !== 'http:' && imageProtocol !== 'https:') {
+      throw new Error('product.image must be served over http/https');
```
Comment (andymckay) [inline src/index.js]: Feels like it should only be http:// for -dev, can we use the httpsOnly setting here?

**Comment by andymckay** [comment]
r+wc

File: src/index.js (hunk 1 above)
Comment (muffinresearch) [inline src/index.js]: ok, I'll add it to the client.

File: src/index.js (hunk 2 above)
Comment (muffinresearch) [inline src/index.js]: The httpsOnly currently relates more to where our content is hosted. But I guess it should also apply to window.location.protocol as well as the image setting too.

**Pull request merged**